client/lib/nav-utils.js

```diff
@@ -15,13 +16,28 @@ export const getAdminLink = path => {
 	return wcSettings.adminUrl + path;
 };
 
-/* Updates the query parameters of the current page.
+/**
+ * Return a URL with set query parameters.
  *
- * @param {Object} Query parameters to be updated.
+ * @param {String} path Relative path (defaults to current path).
+ * @param {Object} currentQuery object of current query params (defaults to current querystring).
+ * @param {Object} query object of params to be updated.
+ * @return {String}  Updated URL merging query params into existing params.
+ */
+export const getNewPath = ( path, currentQuery, query ) => {
+	path = path || history.location.pathname;
+	currentQuery = currentQuery || parse( history.location.search.substring( 1 ) );
+	const queryString = stringifyQueryObject( Object.assign( currentQuery, query ) );
+	return `${ path }?${ queryString }`;
+};
```

Comment:
Great abstraction, thank you. Here's another place it should be used in the date-picker

https://github.com/woocommerce/wc-admin/blob/b7dc5d19751bc6edf73b2a67d9fa3a6af531b327/client/components/date-picker/index.js#L70